x-pack/plugins/security_solution/server/lib/detection_engine/routes/rules/perform_bulk_action_route.ts

```diff
@@ -145,9 +187,51 @@ export const performBulkActionRoute = (
               },
               body: responseBody,
             });
+          case BulkAction.update:
+            processingResponse = await chunkifyRulesAction(rules.data, async (rule) => {
+              throwHttpError(await mlAuthz.validateRuleType(rule.params.type));
+
+              const updatedRule = body.update.reduce(
+                (acc, action) => appplyBulkActionUpdateToRule(acc, action),
+                transformAlertToRule(rule)
+              );
```

Comment:
That's interesting point, thank you @banderror 

I've designed it's with an idea that array actions just aggregations of possible single requests.
So either you can do something like this

POST update/add_tags
POST update/remove_tags
POST update/set_tags

OR that would be equal to

POST [update/add_tags, update/remove_tags, update/set_tags]

In both situations, result would be the same. It's predictable as actions are applying in historical order, and can rewrite each other, which is fine in my opinion as order is explicit and result should be intuitive.

On the other hand, defining order of applied operations by priority can be not very explicit, users will need to have priority map in mind when constructing requests.

Issue here could be only when user starts to put the same value in `add_tags`, `remove_tags`, or use these 2 actions together with `set_tags`. Maybe worth cover this in documentation that order of actions are applying in historical order

Reply:
> In both situations, result would be the same. It's predictable as actions are applying in historical order, and can rewrite each other, which is fine in my opinion as order is explicit and result should be intuitive.
> On the other hand, defining order of applied operations by priority can be not very explicit, users will need to have priority map in mind when constructing requests.

That makes total sense, I agree with you @vitaliidm 👍 Forget my suggestion 🙂 

> Maybe worth cover this in documentation that order of actions are applying in historical order

Hmm, probably yes 🤔 If we document that individual edits are applied in the order of specifying them in the request body, it will bring more clarity for the users. On the other hand, we will have to stick to this behaviour because otherwise changing it later would represent a breaking change in the documented behaviour. Which option are you leaning towards?

Reply:
I think it worth to document it as we give users an opportunity to apply multiple edit actions per request, which can save traffic, time needed to apply bulk edit and simplify code.

Reply:
Sounds good to me 👍 In that case, let's please add it to the description of the endpoint in https://github.com/elastic/security-docs/pull/1317